torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -189,7 +189,10 @@ class ValidateAllocation : private kir::IrVisitor {
     const auto gpu_lower = GpuLower::current();
     for (const auto& allocations : live_allocations_) {
       for (const auto& allocate : allocations) {
-        const auto tv = allocate->buffer()->as<kir::TensorView>();
```

Comment:
nitpick. should we use `isA` here as the condition for short-cut?

Reply:
I generally prefer `dynamic_cast` and nullptr check over `isA` followed by `as` since the latter involves two `dynamic_cast` operations. In terms of code readability, I do prefer the latter approach as it seems more intuitive to me.